src/haste.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-var */
+
+var fs = require('fs');
+var glob = require('glob');
+var path = require('path');
+var _ = require('underscore');
+var perfy = require('perfy');
+
+var providesRegex = /\r?\n \* @providesModule (\S+)(?=\r?\n)/;
+var validName = /^[a-z0-9-_].+$/i;
+var iosTest = /ios/gi;
+var androidTest = /android/gi;
+var CWD = process.cwd();
+var PROJECT_ROOT = path.join(CWD, '..', '..');
+var PROJECT_NODE_MODULES = path.join(PROJECT_ROOT, 'node_modules');
+var TIMER = 'time';
+
+if (!fs.existsSync(PROJECT_NODE_MODULES)) {
+  PROJECT_NODE_MODULES = path.join(CWD, 'node_modules'); // For tests
+}
+
+perfy.start(TIMER);
+
+var data = {
+  hasteMap: {},
+  version: require('./react-native-version')
+};
+
+var files = glob.sync(path.join(PROJECT_NODE_MODULES, '**/*.js'));
+
+_.forEach(files, function (file) {
+  var matches = providesRegex.exec(fs.readFileSync(file).toString());
```

Comment:
I ran into an issue here where if the package name ends with `.js` ([example](https://github.com/whitequark/ipaddr.js)) it will cause the install script to fail. So my workaround was to check if the file is a actual file first. Additionally  I changed the wildcard above to `*/**/*.js`.